addons/hw_scale/controllers/main.py

```diff
@@ -175,6 +213,11 @@ def _parse_weight_answer(self, protocol, answer):
         weight, weight_info, status = None, None, None
         try:
             _logger.debug("Parsing weight [%r]", answer)
+            if not protocol.commandSupport:
```

Comment:
I think you should be able to write the corresponding regex, and avoid this if block